packages/broker/src/plugins/operator/MaintainTopologyHelper.ts

```diff
@@ -34,15 +34,13 @@ export class MaintainTopologyHelper extends EventEmitter<MaintainTopologyHelperE
     private readonly streamIdOfSponsorship: Map<EthereumAddress, StreamID> = new Map()
     private readonly sponsorshipCountOfStream: Map<StreamID, number> = new Map()
     private readonly operatorContractAddress: EthereumAddress
-    private readonly provider: Provider
     private readonly operatorContract: Operator
     private readonly theGraphClient: TheGraphClient
 
-    constructor({ operatorContractAddress, provider, theGraphUrl }: OperatorServiceConfig) {
+    constructor({ operatorContractAddress, nodeWallet, theGraphUrl }: OperatorServiceConfig) {
         super()
         this.operatorContractAddress = operatorContractAddress
-        this.provider = provider
-        this.operatorContract = new Contract(operatorContractAddress, operatorABI, this.provider) as unknown as Operator
+        this.operatorContract = new Contract(operatorContractAddress, operatorABI, nodeWallet) as unknown as Operator
```

Comment:
Just a note here: we don't strictly need a `Signer` compatible interface here. `Provider` is enough since we do read-only operations. Nevertheless, since `OperatorServiceConfig` already has a `Wallet` may as well use that.

Reply:
Good point 👍